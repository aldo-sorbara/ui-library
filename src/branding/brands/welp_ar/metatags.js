export const presto = {
  title: 'Presto Hoy Prestamos Online',
  description: 'Préstamo confiable, rápido y seguro',
  keywords: 'Préstamo, Crédito, Minicrédito, Presto Hoy, Efectivo, Cash'
};
export const posta = {
  title: 'Créditos Posta',
  description: 'Créditos Posta',
  keywords: 'Préstamo, Crédito, Posta, Efectivo, Cash'
};
export const welp = {
  title: 'W E L P',
  description: 'WELP',
  keywords: 'Préstamo, Crédito, Welp, Efectivo, Cash'
};
export default {
  title: 'Onboarding Title',
  description: 'Onboarding Description',
  keywords: 'Onboarding, React, Redux, Styled'
};
