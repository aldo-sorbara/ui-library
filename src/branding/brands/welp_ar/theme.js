import alerts from './jsons/alerts.json';
import assetsPaths from './jsons/assets.json';
import breakPoints from '../common/jsons/breakPoints.json';
import font from './jsons/font.json';
import palette from './jsons/palette.json';
import reset from './jsons/reset.json';
import spacer from './jsons/spacer.json';
import typography from './jsons/typography.json';
import zIndex from '../common/jsons/zIndex.json';

import buttons from './jsons/buttons.json';
import radioButton from './jsons/radioButton.json';
import list from './jsons/list.json';
import containers from './jsons/containers.json';
import modals from './jsons/modals.json';

export default {
  name: 'welp_ar',
  alerts,
  assetsPaths,
  breakPoints,
  buttons,
  list,
  containers,
  emSize: '16px',
  font,
  modals,
  palette,
  radioButton,
  reset,
  spacer,
  typography,
  zIndex
};
