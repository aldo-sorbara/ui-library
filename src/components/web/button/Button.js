import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

/**
 * @render react
 * @name Button
 * @description Button Component for Wenance.
 * @example
 * <ButtonContainer>
 *   <Button data="button"> Button</Button>
 * </ButtonContainer>
 */

export const Button = props => {
  const { children, data, href } = props;
  const Component = href ? StyledLink : StyledButton;
  return (
    <Component {...props} data-w={data}>
      {children}
    </Component>
  );
};

Button.propTypes = {
  children: PropTypes.node.isRequired,
  data: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  isBlocked: PropTypes.bool,
  modifiers: PropTypes.arrayOf(PropTypes.string)
};

Button.defaultProps = {
  disabled: false,
  isBlocked: false,
  modifiers: []
};

const CommonButton = props => `
  border-radius: ${props.theme.buttons.borderRadius};
  display: ${props.isBlocked ? 'block' : 'inline-block'};
  cursor: pointer;
  font-family: inherit;
  opacity: ${props.disabled ? '0.3' : '1'};
  padding: ${props.theme.spacer.medium} ${props.theme.spacer.wrapperInner};
  text-align: center;
  transition: background-color 0.4s ease-in-out;
  touch-action: manipulation;
  user-select: none;
  vertical-align: middle;
  white-space: nowrap;
  width: ${props.isBlocked ? '100%' : 'auto'};
  cursor: ${props.disabled ? 'not-allowed' : 'pointer'};
  background-color: ${props.theme.palette.primary.default};
  border: 1px solid transparent;
  color: ${props.theme.palette.common.text.inverse};

  &:hover:enabled {
    background-color: ${props.theme.palette.primary.dark};
  }

  &:focus {
    outline: none;
  }

  &:disabled,
  button[disabled] {
    opacity: 0.3;
  }
`;

CommonButton.defaultProps = {
  theme: {
    palette: {
      common: {
        text: {
          inverse: '#ffffff'
        }
      },
      primary: {
        default: '#0082ff',
        dark: '#3a66ab'
      }
    },
    typography: {
      button: {
        fontSize: '.875rem',
        fontWeight: '700',
        lineHeight: '1.145'
      }
    },
    spacer: {
      medium: '1rem',
      large: '1.5rem',
      button: {
        smallSpacer: '.75rem'
      }
    },
    buttons: {
      borderRadius: '0'
    }
  }
};

const StyledButton = styled.button`
  ${CommonButton}
  ${props => props.theme.typography.button};
  ${props => props.modifiers.map(modifier => modifiers[modifier])}
`;

const StyledLinkButton = props => `
  background: transparent;
  border: 1px solid transparent;
  color: ${props.theme.palette.primary.default};
  &:hover, &:hover:enabled {
    background-color: inherit;
    opacity: .65
  }`;

const StyledLink = styled.a`
  ${CommonButton}
  ${props => props.theme.typography.button};
  ${props => props.modifiers.map(modifier => modifiers[modifier])}
  text-decoration: none;
`;

const StyledSmallButton = props => `
  padding: ${props.theme.spacer.button.smallSpacer} ${props.theme.spacer.large};
`;

const StyledInverseButton = props => `
  background: transparent;
  color: ${props.theme.palette.common.text.inverse};
  border: 1px solid ${props.theme.palette.common.text.inverse};
  
  &:hover, &:hover:enabled {
    background-color: inherit;
    opacity: .65
  }
`;

const StyledRoundedButton = () => `
  border-radius: 50%;
  width: 50px;
  height: 50px
  padding: 0;

  svg {
    display: block;
    margin: auto;
  }
`;

const modifiers = {
  small: StyledSmallButton,
  rounded: StyledRoundedButton,
  link: StyledLinkButton,
  inverse: StyledInverseButton
};

// ButtonContainer Component
export const ButtonContainer = props => {
  return (
    <StyledButtonContainer {...props}>
      <div>{props.children}</div>
    </StyledButtonContainer>
  );
};

const StyledButtonContainer = styled.div`
  padding: ${props => props.theme.buttons.padding};
`;

StyledButtonContainer.defaultProps = {
  theme: {
    buttons: {
      padding: '0 2rem 2rem'
    }
  }
};
