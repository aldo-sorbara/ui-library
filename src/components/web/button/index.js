export { Button, ButtonContainer } from './Button';
