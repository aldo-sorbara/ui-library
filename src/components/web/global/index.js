export {
  Container,
  Wrapper,
  Headline,
  Description,
  Legales,
  Label,
  LabelLarge,
  SecondHeadline,
  ContainerDesktop,
  Desktop,
  Onboarding,
  Terms,
  H1,
  SectionDescription,
  UrlLink,
  Timer,
  DesktopText,
  Bind,
  Logo,
  LogoMobile,
  TextPromotion
} from './Global';
