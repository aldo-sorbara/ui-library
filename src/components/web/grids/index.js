export { Grid, Container as GridContainer } from './Grids';
