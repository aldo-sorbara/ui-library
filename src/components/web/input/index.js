export { default, InputLabel, InputDescription } from './Input';
