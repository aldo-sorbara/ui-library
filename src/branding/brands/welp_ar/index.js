import wording from './jsons/wording.json';
import theme from './theme';
import config from './jsons/config.json';
import feedback from './jsons/feedback.json';
import assets from './jsons/assets.json';
import palette from './jsons/palette.json';

export default {
  wording: { onboarding: wording },
  feedback: { onboarding: feedback },
  theme,
  config: { onboarding: config },
  typography: theme.font.fontfamily,
  assets: { onboarding: assets },
  palette
};
