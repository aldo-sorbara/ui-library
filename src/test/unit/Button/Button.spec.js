import React from 'react';
import { Button, ButtonContainer } from '../../../components/web/button';
import renderer from 'react-test-renderer';
import { mountWithTheme } from '../../../../utils/helper';
import welp_ar from '../../../branding/brands/welp_ar';
import { ThemeProvider } from 'styled-components';

import 'jest-styled-components';

describe('Button component', () => {
  it('should render snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <Button className='test-button' data='click-me'>
          Click me
        </Button>
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render a disabled button', () => {
    const wrapper = mountWithTheme(
      <Button className='test-button' disabled data='disabled'>
        Disabled Button
      </Button>,
      welp_ar.theme
    );
    expect(wrapper.props().disabled).toBe(true);
    expect(wrapper).toHaveStyleRule('cursor', 'not-allowed');
    expect(wrapper).toHaveStyleRule('opacity', '0.3');
  });
  it('should render a block button', () => {
    const wrapper = mountWithTheme(
      <Button className='test-button' isBlocked data='isBlocked'>
        Button isBlocked
      </Button>,
      welp_ar.theme
    );
    expect(wrapper.props().isBlocked).toBe(true);
    expect(wrapper).toHaveStyleRule('display', 'block');
    expect(wrapper).toHaveStyleRule('width', '100%');
  });
  it('should render ButtonContainer snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <ButtonContainer
          className='test-buttonContainer'
          data='button-container'
        >
          This is a button container
        </ButtonContainer>
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call button onClick', () => {
    const onClickCallback = jest.fn();
    const wrapper = mountWithTheme(
      <Button onClick={onClickCallback} className='test-button' data='click'>
        Button Click
      </Button>,
      welp_ar.theme
    );
    wrapper
      .find('button')
      .at(0)
      .simulate('click');
    expect(onClickCallback).toHaveBeenCalled();
  });
  it('should render a link', () => {
    const wrapper = mountWithTheme(
      <Button href='#' className='test-button' data='link'>
        Link Button
      </Button>,
      welp_ar.theme
    );
    expect(wrapper.props().href).toBe('#');
    expect(
      wrapper
        .find('a')
        .at(0)
        .exists()
    ).toEqual(true);
  });
  it('should render a button with link modifier', () => {
    const modifiers = ['link'];
    const wrapper = mountWithTheme(
      <Button
        href='#'
        modifiers={modifiers}
        className='test-button'
        data='link'
      >
        Link Button
      </Button>,
      welp_ar.theme
    );
    expect(wrapper.props().href).toBe('#');
    expect(wrapper.props().modifiers).toEqual(
      expect.arrayContaining(modifiers)
    );
    expect(
      wrapper
        .find('a')
        .at(0)
        .exists()
    ).toEqual(true);
    expect(wrapper).toHaveStyleRule('background', 'transparent');
    expect(wrapper).toHaveStyleRule('border', '1px solid transparent');
  });
  it('should render a button with small modifier', () => {
    const modifiers = ['small'];
    const wrapper = mountWithTheme(
      <Button modifiers={modifiers} className='test-button' data='small'>
        small Button
      </Button>,
      welp_ar.theme
    );
    expect(wrapper.props().modifiers).toEqual(
      expect.arrayContaining(modifiers)
    );
    expect(wrapper).toHaveStyleRule('padding', '.75rem 1.5rem');
  });
  it('should render a button with rounded modifier', () => {
    const modifiers = ['rounded'];
    const wrapper = mountWithTheme(
      <Button modifiers={modifiers} className='test-button' data='rounded'>
        rounded Button
      </Button>,
      welp_ar.theme
    );
    expect(wrapper.props().modifiers).toEqual(
      expect.arrayContaining(modifiers)
    );
    expect(wrapper).toHaveStyleRule('border-radius', '50%');
    expect(wrapper).toHaveStyleRule('width', '50px');
    expect(wrapper).toHaveStyleRule('height', '50px');
    expect(wrapper).toHaveStyleRule('padding', '0');
  });
  it('should render a button with inverse modifier', () => {
    const modifiers = ['inverse'];
    const wrapper = mountWithTheme(
      <Button modifiers={modifiers} className='test-button' data='inverse'>
        inverse Button
      </Button>,
      welp_ar.theme
    );
    expect(wrapper.props().modifiers).toEqual(
      expect.arrayContaining(modifiers)
    );
    expect(wrapper).toHaveStyleRule('border', '1px solid #ffffff');
    expect(wrapper).toHaveStyleRule('color', '#ffffff');
  });
});
