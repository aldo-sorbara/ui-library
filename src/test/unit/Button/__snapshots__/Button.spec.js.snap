// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render ButtonContainer snapshot 1`] = `
.c0 {
  padding: 0 2rem 2rem;
}

<div
  className="test-buttonContainer c0"
  data="button-container"
>
  <div>
    This is a button container
  </div>
</div>
`;

exports[`Button component should render snapshot 1`] = `
.c0 {
  border-radius: undefined;
  display: inline-block;
  cursor: pointer;
  font-family: inherit;
  opacity: 1;
  padding: 1rem 2rem;
  text-align: center;
  -webkit-transition: background-color 0.4s ease-in-out;
  transition: background-color 0.4s ease-in-out;
  touch-action: manipulation;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  white-space: nowrap;
  width: auto;
  cursor: pointer;
  background-color: #0082ff;
  border: 1px solid transparent;
  color: #ffffff;
  font-size: .875rem;
  font-weight: 700;
  line-height: 1.14;
}

.c0:hover:enabled {
  background-color: #3a66ab;
}

.c0:focus {
  outline: none;
}

.c0:disabled,
.c0 button[disabled] {
  opacity: 0.3;
}

<button
  className="test-button c0"
  data="click-me"
  data-w="click-me"
  disabled={false}
>
  Click me
</button>
`;
